storage/storage.go

```diff
@@ -241,6 +245,26 @@ type RefreshToken struct {
 	Nonce string
 }
 
+// RefreshTokenRef is a reference object that contains metadata about refresh tokens.
+type RefreshTokenRef struct {
+	ID        string
+	CreatedAt time.Time
+	LastUsed  time.Time
```

Comment:
Changed the `Refresh` attribute to be of type map[string]*RefreshTokenRef indexed by the ClientID of the refresh token. 

Pros: 
* efficient access of object using ClientID of refresh token since it is a hash table indexed by ClientID
* easy update of attributes (like LastUsed) since it a pointer to `RefreshTokenRef`

Also renamed OfflineSession to OfflineSessions as you recommended